### PR TITLE
App Service refactor and feature alignment

### PR DESCRIPTION
**Pull request by jackofallops**
Changes:

* Data Source: `azurerm_app_service` now exports `source_control` configuration
* Data Source: `azurerm_function_app` now exports `source_control` configuration
* Data Source: `azurerm_function_app` now exports `site_config` configuration

* `azurerm_app_service` now supports `source_control` configuration
* `azurerm_function_app` now supports `source_control` configuration
* `azurerm_function_app` now supports full `ip_restriction` configuration
* `azurerm_function_app` now supports full `scm_ip_restriction` configuration

### CODE REVIEW

File: azurerm/internal/services/web/resource_arm_function_app_slot.go [hunk 1]
```diff
@@ -779,117 +694,117 @@ func expandFunctionAppSlotAppSettings(d *schema.ResourceData, appServiceTier, en
...
-	result["cors"] = azure.FlattenWebCorsSettings(input.Cors)
-
-	if input.AutoSwapSlotName != nil {
-		result["auto_swap_slot_name"] = *input.AutoSwapSlotName
-	}
-
-	results = append(results, result)
-	return results
-}
+//func expandFunctionAppSlotSiteConfig(d *schema.ResourceData) (web.SiteConfig, error) {
```
Comment (katbyte) [inline azurerm/internal/services/web/resource_arm_function_app_slot.go:697]: This should be removed ?

File: azurerm/internal/services/web/resource_arm_function_app_slot.go [hunk 2]
```diff
@@ -911,45 +826,45 @@ func expandFunctionAppSlotConnectionStrings(d *schema.ResourceData) map[string]*
...
-		if vNetSubnetID != "" {
-			ipSecurityRestriction.VnetSubnetResourceID = &vNetSubnetID
-		}
-
-		restrictions = append(restrictions, ipSecurityRestriction)
-	}
-
-	return restrictions, nil
-}
+//func expandFunctionAppSlotIPRestriction(input interface{}) ([]web.IPSecurityRestriction, error) {
```
Comment (katbyte) [inline azurerm/internal/services/web/resource_arm_function_app_slot.go:829]: and this?

File: azurerm/internal/services/web/resource_arm_function_app_slot.go [hunk 3]
```diff
@@ -985,32 +900,32 @@ func flattenFunctionAppSlotSiteCredential(input *web.UserProperties) []interface
...
-
-		restrictions = append(restrictions, map[string]interface{}{
-			"ip_address": ipAddress,
-			"subnet_id":  subnetID,
-		})
-	}
-
-	return restrictions
-}
+//func flattenFunctionAppSlotIPRestriction(input *[]web.IPSecurityRestriction) []interface{} {
```
Comment (katbyte) [inline azurerm/internal/services/web/resource_arm_function_app_slot.go:903]: and this?

File: azurerm/internal/services/web/tests/resource_arm_app_service_slot_test.go [hunk 4]
```diff
@@ -2799,6 +2837,123 @@ resource "azurerm_app_service_slot" "test" {
...
 }
 
+func testAccAzureRMAppServiceSlot_scmUseMainIPRestriction(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
```
Comment (katbyte) [inline azurerm/internal/services/web/tests/resource_arm_app_service_slot_test.go:2847]:
```suggestion
  name     = "acctestRG-web-%d"
```

File: azurerm/internal/services/web/tests/resource_arm_app_service_slot_test.go [hunk 5]
```diff
@@ -2799,6 +2837,123 @@ resource "azurerm_app_service_slot" "test" {
...
+}
+
+func testAccAzureRMAppServiceSlot_scmOneIPRestriction(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
```
Comment (katbyte) [inline azurerm/internal/services/web/tests/resource_arm_app_service_slot_test.go:2904]:
```suggestion
  name     = "acctestRG-web-%d"
```

File: azurerm/internal/services/web/tests/resource_arm_app_service_test.go [hunk 6]
```diff
@@ -3981,6 +4024,44 @@ resource "azurerm_app_service" "test" {
...
 }
 
+func testAccAzureRMAppService_withSourceControl(data acceptance.TestData, branch string) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
```
Comment (katbyte) [inline azurerm/internal/services/web/tests/resource_arm_app_service_test.go:4034]: we should include the service here
```suggestion
  name     = "acctestRG-web-%d"
```

**Comment by jackofallops** [comment]
Tests Passing:
image
Failures transient

**Pull request merged**